semantickernel-api/src/main/java/com/microsoft/semantickernel/hooks/KernelHooks.java

```diff
@@ -224,6 +225,25 @@ public boolean isEmpty() {
         return hooks.isEmpty();
     }
 
+    /**
+     * Builds the list of hooks to be invoked for the given context, by merging the hooks in this
+     * collection with the hooks in the context. Hooks in the context will override hooks in this.
+     *
+     * @param context the invocation context
+     * @return the list of hooks to be invoked
+     */
+    public KernelHooks buildInvocationHooks(@Nullable InvocationContext context) {
```

Comment:
Why not just pass `KernelHooks other`? Wouldn't that be more flexible for, say, merging the hooks from another Kernel (or building with hooks from another Kernel).